Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/RCFCaster.java

```diff
@@ -246,6 +247,34 @@ public TimedRangeVector extrapolate(Calibration calibration, int horizon, boolea
 
     @Override
     public List<AnomalyDescriptor> processSequentially(double[][] data, Function<AnomalyDescriptor, Boolean> filter) {
+        if (data == null || data.length == 0) {
+            return new ArrayList<>();
+        }
+
+        long timestamp = preprocessor.getInternalTimeStamp();
+        long[] timestamps = new long[data.length];
+        for (int i = 0; i < data.length; i++) {
+            timestamps[i] = ++timestamp;
```

Comment:
are we just internally increasing timestamp by one for each next data point? Is that a correct assumption?

Reply:
oh nevermind I see this is what we basically did before but now we are allowing other callers to pass in specific timestamps themselves

Reply:
yes